src/diffusers/models/attention_processor.py

```diff
@@ -1129,11 +1129,27 @@ def __call__(
             if attn.norm_added_k is not None:
                 encoder_hidden_states_key_proj = attn.norm_added_k(encoder_hidden_states_key_proj)
 
+            query_pad_size = query.size(2)
+            key_pad_size = key.size(2)
+
             query = torch.cat([query, encoder_hidden_states_query_proj], dim=2)
             key = torch.cat([key, encoder_hidden_states_key_proj], dim=2)
             value = torch.cat([value, encoder_hidden_states_value_proj], dim=2)
 
-        hidden_states = F.scaled_dot_product_attention(query, key, value, dropout_p=0.0, is_causal=False)
+            if attention_mask is not None:
```

Comment:
well I think what we want is not to have a specific implementation to apply attention_mask for flux, is just to allow it to pass down all the way from pipeline, to transformer and the to attention processor so user can experiment with a custom attention mask 

cc @christopher5106 is what I described here something you have in mind?

Reply:
and if you do have an specific implementation that we want to add to diffusers,  maybe you can run some experiments to help us decide if it's meaningful

Reply:
it's the encoder attention mask though just following convention of pixart and other DiT that rely on attention masking. masking the attention arbitrarily doesn't unlock new use cases, does it? if so, providing examples of those would be nice.

Reply:
@rootonchair yes feel free to modify pipeline/model to test, and provide the experiments results to us:)
>how should I test this feature? modify the original flux pipeline?

Reply:
cc @christopher5106 - would you be able to provide a use case? since it was the original ask
> masking the attention arbitrarily doesn't unlock new use cases, does it? if so, providing examples of those would be nice.

Reply:
checking the softmax scores for padded positions.